pkgs/development/libraries/paho-mqtt-c/default.nix

```diff
@@ -0,0 +1,34 @@
+{ lib, stdenv, fetchFromGitHub, cmake, openssl, makeWrapper }:
+
+stdenv.mkDerivation rec {
+  pname = "paho.mqtt.c";
+  version = "1.3.11";
+
+  src = fetchFromGitHub {
+    owner = "eclipse";
+    repo = "paho.mqtt.c";
+    rev = "v${version}";
+    hash = "sha256-TGCWA9tOOx0rCb/XQWqLFbXb9gOyGS8u6o9fvSRS6xI=";
+  };
+
+  nativeBuildInputs = [ cmake makeWrapper ];
+
+  buildInputs = [ openssl ];
+
+  cmakeFlags = [ "-DPAHO_WITH_SSL=TRUE" ];
+
+  postFixup = ''
+    # for dlopen
+    wrapProgram $out/bin/MQTTVersion \
+      --prefix LD_LIBRARY_PATH : "$out/lib" \
+      --prefix DYLD_LIBRARY_PATH : "$out/lib"
```

Comment:
How many dlopen calls are there? If there are just a handfull we could patch them or rather use patchelf.

Reply:
It tries to dlopen 4 libraries:
https://github.com/eclipse/paho.mqtt.c/blob/7c0735a5f02052c275baddb4f01f1f4dabdfed20/src/MQTTVersion.c#L57